File: common-lib/src/main/java/club/hosppy/common/validation/PhoneNumberValidator.java
package club.hosppy.common.validation;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

public class PhoneNumberValidator implements ConstraintValidator<PhoneNumber, String> {

    @Override
    public boolean isValid(String phoneNumber, ConstraintValidatorContext context) {
        return phoneNumber == null || phoneNumber.matches("[0-9]{9,13}");
    }
}

File: src/main/kotlin/club/hosppy/common/validation/PhoneNumberValidator.kt
package club.hosppy.common.validation

import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class PhoneNumberValidator : ConstraintValidator<PhoneNumber?, String?> {
    override fun isValid(phoneNumber: String?, context: ConstraintValidatorContext): Boolean {
        return phoneNumber == null || phoneNumber.matches(Regex("[0-9]{9,13}"))
    }
}